.github/workflows/linux-check.yaml

```diff
@@ -63,7 +63,7 @@ jobs:
           CXX: clang++-14
         # -g1 should slightly reduce build time.
         run: |
-          cmake . -B build -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS=-g1 -DUNITY_DISABLE=ON
+          cmake . -B build -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS=-g1 -DUNITY_DISABLE=ON -DUSE_CLANG_TIDY=ON
```

Comment:
It was just most "convenient" place in current CI configuration to put this flag, but it is not a big deal to do it in another way. We can:
- Extract it to separate job
- Move to some another job
- Leave it here
- ...
Let's just discuss, I am not sure what is the best solution.

Reply:
> Can it be run independently on its own, without building the whole project?

Yes, but it needs an up to date compile command database, where for each and every compilation unit, the info about how exactly to build is stored. Like where the includes are coming from, what is defined etc.
And therfore it is typical, to run it together with the compilation, so the result is surely not out of date.

BTW, if we are into static checking, then there is [this frontend](https://github.com/Ericsson/codechecker#command-line-cc-analysis), which next to `clang-tidy` could do even more, if needed..

Reply:
@Ferenc- if we run it independently is there good way to control for which targets we run it and for which not? E.g. I don’t think we want to run it for third party dependencies and CXX_CLANG_TIDY allows to control it quite easily(we just don’t set it for targets we don’t want to run clang-tidy for and that’s it)

Reply:
@Ferenc- did you use codechecker before? Does it integrate easily with CMake?
Can't compile command database be generated _without_ building the binaries? My main point is to do as fast check stage as possible, without unnecessary build overhead.

Reply:
It seems we can just set this variable and it will output this database right after CMake https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html (but it is to be checked - will do when will have a time)